src/Evolution/Systems/GrMhd/ValenciaDivClean/KastaunEtAlHydro.cpp

```diff
@@ -46,6 +46,17 @@ class FunctionOfZ {
         electron_fraction_(electron_fraction),
         equation_of_state_(equation_of_state) {
     r_ = std::sqrt(r_squared_);
+
+    // Internal consistency check
+
+    if constexpr (EnforcePhysicality) {
```

Comment:
(continued from the last comment)

but here in the `hpp` file `EnforcePhysicality` was set to `true`. I think the enforcing physicality here puts the insane value of momentum density
```
get<0>(*tilde_s_ptr)[point_to_change] = 1e3;
```
(line 213 of Test_TciOnDgGrid.cpp`) back to a correct physical value, then the prim recovery actually succeeds.

Reply:
If my guess is correct, we need to fix the line 213 of `Test_TcionDgGrid` so that it puts another, well-failing (?) combination of conserved variables so that the prim recovery can successfully fail.

Reply:
Yep, that's exactly what happens. Currently the routine fixes too aggressively, when it's not supposed to do that. I added a missing commit that should address this.